uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/test/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHServiceTest.java

```diff
@@ -260,9 +282,287 @@ public void testCheckSetupParameters() throws Exception {
+        final File certDir = createTempDirectory();
+
+        String idleTimeout = "10000";
+        sshService.setup(certDir,
+                         null,
+                         idleTimeout,
+                         "RSA",
+                         mock(ReceivePackFactory.class),
+                         mock(JGitFileSystemProvider.RepositoryResolverImpl.class),
+                         executorService,
+                         "",
+                         "");
+
+        sshService.start();
+        assertTrue(sshService.isRunning());
+
+        List<String> ciphersReaded = sshService.getSshServer().getCipherFactoriesNames();
+        List<String> macsReaded = sshService.getSshServer().getMacFactoriesNames();
+
+        assertThat(ciphersReaded).hasSize(7);
+        assertThat(ciphersReaded.containsAll(Arrays.asList("aes128-ctr",
+                                                           "aes192-cbc",
+                                                           "aes192-ctr",
+                                                           "aes256-cbc",
+                                                           "aes256-ctr",
+                                                           "arcfour128",
+                                                           "arcfour256")));
+        assertThat(macsReaded).hasSize(6);
+        assertThat(macsReaded.containsAll(Arrays.asList("HmacMD5",
+                                                        "HmacMD5",
+                                                        "HmacSHA1",
+                                                        "HmacSHA1",
+                                                        "HmacSHA256",
+                                                        "HmacSHA512")));
+
+        assertThat(sshService.getSshServer().getProperties().get(SshServer.IDLE_TIMEOUT)).isEqualTo(idleTimeout);
+
+        sshService.stop();
+
+        assertFalse(sshService.isRunning());
+    }
+
+    @Test
+    public void testCheckNullCiphersAndMacs() throws Exception {
+        final GitSSHService sshService = new GitSSHService();
+        final File certDir = createTempDirectory();
+
+        String idleTimeout = "10000";
+        sshService.setup(certDir,
+                         null,
+                         idleTimeout,
+                         "RSA",
+                         mock(ReceivePackFactory.class),
+                         mock(JGitFileSystemProvider.RepositoryResolverImpl.class),
+                         executorService,
+                         null,
+                         null);
+
+        sshService.start();
+        assertTrue(sshService.isRunning());
+
+        List<String> ciphersReaded = sshService.getSshServer().getCipherFactoriesNames();
+        List<String> macsReaded = sshService.getSshServer().getMacFactoriesNames();
+
+        assertThat(ciphersReaded).hasSize(7);
+        assertThat(ciphersReaded.containsAll(Arrays.asList("aes128-ctr",
```

Comment:
this test is fragile, better before iterate over all available ciphers and macs to make sure that in a upgrade of library we don't break the test

Reply:
Ok